core/src/Platforms/iOS/WebviewBase.cs

```diff
@@ -60,5 +64,36 @@ public static bool ContinueAuthentication(string url)
 
             return true;
         }
+
+        protected void OnMoveToBackground(NSNotification notification)
+        {
+            // After iOS 11.3, it is neccesary to keep a background task running while moving an app to the background 
+            // in order to prevent the system from reclaiming network resources from the app. 
+            // This will prevent authentication from failing while the application is moved to the background while waiting for MFA to finish.
+            taskId = UIApplication.SharedApplication.BeginBackgroundTask(() =>
+            {
+                if (taskId != UIApplication.BackgroundTaskInvalid)
+                {
+                    UIApplication.SharedApplication.EndBackgroundTask(taskId);
+                    taskId = UIApplication.BackgroundTaskInvalid;
+                }
+            });
+        }
+
+        protected void OnMoveToForeground(NSNotification notification)
+        {
+            if (taskId != UIApplication.BackgroundTaskInvalid)
+            {
+                UIApplication.SharedApplication.EndBackgroundTask(taskId);
+                taskId = UIApplication.BackgroundTaskInvalid;
+            }
+        }
+
+        //Hiding NSObject.Dispose() with new to implement IDisposable interface
+        public new void Dispose()
```

Comment:
>new [](start = 15, length = 3)

the usual dispose pattern... I believe has an overload which takes a bool which can be overridden. You should probably use that one, otherwise you are not guaranteed the dispose handler is called at the expected time (unless the one calling does so on the actual type)

Reply:
sounds good. thanks for the feedback.